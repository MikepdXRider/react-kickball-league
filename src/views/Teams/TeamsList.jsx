import React, { useEffect, useState } from 'react'
import TeamCard from '../../components/TeamCard/TeamCard.jsx';
import { getTeams } from '../../services/teams.jsx';

export default function TeamsList() {
    // establish state hooks
    const [teamsDataArr, setTeamsDataArr] = useState([]);
    const [isLoading, setIsLoading] = useState(false);

    // use useEffect hooks to manage API requests.
    useEffect(() => {
        setIsLoading(true);
        async function getTeamsData(){
            const newTeamsDataArr = await getTeams();
            console.log('getTeams fn returns :', newTeamsDataArr);
            setTeamsDataArr(newTeamsDataArr);
        }

        getTeamsData();
        setIsLoading(false);
    }, [])

    return (
        <main>
            {
                isLoading 
                ? <h1>Loading...</h1> 
                : <section>
                    <ul>
                        {teamsDataArr.map(teamDataObj => {
                            return (
                                <li key={teamDataObj.id}>
                                    <TeamCard teamDataObj={teamDataObj}/>
                                </li>
                            )
                        })}
                    </ul>
                </section>
            }
        </main>
    )
}
