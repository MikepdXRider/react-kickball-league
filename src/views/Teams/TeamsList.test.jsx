import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import TeamsList from "./TeamsList.jsx";


it ('renders TeamsList page', async () => {
    const { container } = render(
            <MemoryRouter>
                <TeamsList />
            </MemoryRouter>
        )
    // count nest line 14 in waitFor() - maybe. 
    const teamName = await screen.findByText('Acme USA');

    expect(teamName).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})