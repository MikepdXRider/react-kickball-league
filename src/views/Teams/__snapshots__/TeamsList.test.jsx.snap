// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TeamsList page 1`] = `
<div>
  <main>
    <section>
      <ul>
        <li>
          <a
            href="/teams/10"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Acme USA
              </h3>
              <h6>
                2021-11-29T01:33:23.387367+00:00
              </h6>
              <h5>
                Anytown, MN 
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/teams/1"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Bridge City Sneakers
              </h3>
              <h6>
                2021-11-22T20:07:58+00:00
              </h6>
              <h5>
                Portland, OR 
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/teams/3"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Lakeville Thunderfeet
              </h3>
              <h6>
                2021-11-22T20:08:43+00:00
              </h6>
              <h5>
                Lake Oswego, OR 
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/teams/4"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Mt. Hoodies
              </h3>
              <h6>
                2021-11-22T20:10:16+00:00
              </h6>
              <h5>
                Mt. Hood, OR 
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/teams/2"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Stumptown Lumberjacks
              </h3>
              <h6>
                2021-11-22T20:08:18+00:00
              </h6>
              <h5>
                Portland, OR 
              </h5>
            </article>
          </a>
        </li>
        <li>
          <a
            href="/teams/5"
          >
            <article
              style="border: 1px solid black;"
            >
              <h3>
                Vancougers
              </h3>
              <h6>
                2021-11-22T23:16:31+00:00
              </h6>
              <h5>
                Vancouver, WA 
              </h5>
            </article>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
